src/core/qgsmapsettings.cpp

```diff
@@ -436,6 +436,36 @@ QgsCoordinateTransform QgsMapSettings::layerTransform( const QgsMapLayer *layer
   return QgsCoordinateTransform( layer->crs(), mDestCRS, mTransformContext );
 }
 
+QgsRectangle QgsMapSettings::computeExtentForScale( const QgsPointXY &center, double scale ) const
+{
+  // Output width in inches
+  double outWIn = outputSize().width() / double( outputDpi() );
+
+  // Desired visible width (honouring scale)
+  double scaledWIn = outWIn * scale;
```

Comment:
scaledWidthInInches? I'm not sure what scaledWIn is short for :stuck_out_tongue: